src/getty-generator/getty-generator.c

```diff
@@ -226,30 +223,24 @@ static int run(const char *dest, const char *dest_early, const char *dest_late)
                 return run_container();
 
         /* Automatically add in a serial getty on all active kernel consoles */
-        _cleanup_free_ char *active = NULL;
-        (void) read_one_line_file("/sys/class/tty/console/active", &active);
-        for (const char *p = active;;) {
-               _cleanup_free_ char *tty = NULL;
-
-                r = extract_first_word(&p, &tty, NULL, 0);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to parse /sys/class/tty/console/active: %m");
-                if (r == 0)
-                        break;
-
-                /* We assume that gettys on virtual terminals are started via manual configuration and do
-                 * this magic only for non-VC terminals. */
-
-                if (isempty(tty) || tty_is_vc(tty))
-                        continue;
+        _cleanup_strv_free_ char **consoles = NULL;
+        r = get_kernel_consoles(&consoles);
+        if (r < 0)
+                log_warning_errno(r, "Failed to get active kernel consoles, ignoring: %m");
+        else if (r > 0)
```

Comment:
This is not necessary. When `r < 0`, `consoles` is `NULL`. When `r == 0`, `consoles` only has "/dev/console", and will be filtered by `tty_is_vc()` below.

Reply:
yeah, but I prefer to be explicit here. there's no need to ever look at `consoles` if r <= 0

Reply:
> When r == 0, consoles only has "/dev/console", and will be filtered by tty_is_vc() below.

That's not really the case though? We're not calling `tty_is_vc_resolved` (and it's unnecessary)?

And it feels nicer to be explicit here, i.e. if r <= 0 there's no need to look at `consoles`